### PR TITLE
Zone name exists

### PR DESCRIPTION
**Pull request by mdb**
Fixes issue #63

### CODE REVIEW

File: vinyldns/integration_test.go [hunk 1]
```diff
@@ -95,6 +95,17 @@ func TestGroupAdminsIntegration(t *testing.T) {
...
+func TestZoneNameExistsForNonexistentZoneIntegration(t *testing.T) {
+	c := client()
+	exists, err := c.ZoneNameExists("foo")
+	if err != nil {
+		t.Error(err)
+	}
+	if exists != false {
+		t.Error(fmt.Sprintf("expected ZoneNameExists to return false; got %t", exists))
+	}
+}
```
Comment (megadeezl) [inline vinyldns/integration_test.go]: 👍

File: vinyldns/integration_test.go [hunk 2]
```diff
@@ -141,6 +152,17 @@ func TestZoneCreateIntegration(t *testing.T) {
...
+func TestZoneNameExistsForExistentZoneIntegration(t *testing.T) {
+	c := client()
+	exists, err := c.ZoneNameExists("ok.")
+	if err != nil {
+		t.Error(err)
+	}
+	if exists != true {
+		t.Error(fmt.Sprintf("expected ZoneNameExists to return true; got %t", exists))
+	}
+}
```
Comment (megadeezl) [inline vinyldns/integration_test.go]: 👍

File: vinyldns/zones.go [hunk 3]
```diff
@@ -123,15 +124,35 @@ func (c *Client) ZoneDelete(zoneID string) (*ZoneUpdateResponse, error) {
...
+	err := resourceRequest(c, zoneEP(c, id), "GET", nil, zone)
 	if err != nil {
-		return false, err
+		if vErr, ok := err.(*Error); ok {
+			if vErr.ResponseCode == http.StatusNotFound {
+				return false, nil
+			}
+
+			return false, err
+		}
```
Comment (megadeezl) [inline vinyldns/zones.go]: 👍

File: vinyldns/zones.go [hunk 4]
```diff
@@ -123,15 +124,35 @@ func (c *Client) ZoneDelete(zoneID string) (*ZoneUpdateResponse, error) {
...
+	zone := &ZoneResponse{}
+	err := resourceRequest(c, zoneNameEP(c, name), "GET", nil, zone)
+	if err != nil {
+		if vErr, ok := err.(*Error); ok {
+			if vErr.ResponseCode == http.StatusNotFound {
+				return false, nil
+			}
+
+			return false, err
+		}
```
Comment (megadeezl) [inline vinyldns/zones.go]: 👍

**Comment by mdb** [comment]
Thanks for the review @megadeezl !

**Pull request merged**